src/main/java/com/google/devtools/build/lib/query2/engine/QueryEnvironment.java

```diff
@@ -159,6 +161,11 @@ public final FilteringQueryFunction asFilteringFunction() {
     public abstract int getExpressionToFilterIndex();
   }
 
+  @FunctionalInterface
+  interface TargetLookup {
+    Target getTarget(Label label) throws TargetNotFoundException, InterruptedException;
```

Comment:
apologies I think I'm missing something - where does this actually get implemented?

Reply:
It's a functional interface, so in ActionGraphQueryEnvironment's constructor, the method reference ActionGraphQueryEnvironment::getTarget is used as a TargetLookup and passed into ConfiguredTargetValueAccessor.

Reply:
Ah, and AGQE has an inherited getTarget method from PostAnalysisQE. Missed that bit. Thanks!